Sources/Testing/Events/Event.swift

```diff
@@ -158,11 +168,23 @@ public struct Event: Sendable {
     instant: Test.Clock.Instant = .now,
     configuration: Configuration? = nil
   ) {
-    // Create both the event and its associated context here at same point, to
-    // ensure their task local-derived values are the same.
-    let event = Event(kind, testID: test?.id, instant: instant)
     let context = Event.Context(test: test, testCase: testCase)
-    event._post(in: context, configuration: configuration)
+
+    withConfiguration(configuration) { configuration in
```

Comment:
This really seems like the wrong way to handle this since the configuration may well not be available or knowable here, and you're using it _just_ for `isTestArgumentEncodingEnabled`. I'd expect that this property would generally be `true`, and having it be unexpectedly `false` (because e.g. an event is posted from a detached task) has spooky effects.